extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -345,6 +347,45 @@ export class AmpA4A extends AMP.BaseElement {
 
     /** @private {string} */
     this.safeframeVersion_ = DEFAULT_SAFEFRAME_VERSION;
+
+    /** @private @const {RefreshManager}  */
+    this.refreshManager_ = getRefreshManagerFor(this.win);
+
+    this.refreshManager_.registerElement(this.element, refresher => {
+      console.log('Eureka!');
+      this.isRefreshing_ = true;
+      this.unlayoutCallback();
```

Comment:
Agreed and doesn't unlayoutCallback remove the frame causing the existing creative to be destroyed?  The idea is to leave the existing creative in place until we have retrieved the new creative so instead of calling unlayoutCallback, you should just create a new instance of the adPromise to "restart" ad retrieval and then on response you can clear or instead have the creative render code detect its in the refresh state and clear the frame just prior to creating it.